parcels/field.py

```diff
@@ -200,7 +200,7 @@ def __init__(self, name, data, lon=None, lat=None, depth=None, time=None, grid=N
             self.units = self.unitconverters[fieldtype]
         else:
             raise ValueError("Unsupported mesh type. Choose either: 'spherical' or 'flat'")
-        self.interp_method = interp_method
+        self.interp_method = interp_method[name] if name in interp_method else interp_method
```

Comment:
interp_method should be tested to see if it is a dict or not

Example that does not work currently: `name = 'i'` and `interp_method = 'linear'`
In this case `name in interp_method` returns `True`, then `interp_method[name]` breaks